nsqd/nsqd.go

```diff
@@ -223,6 +239,27 @@ func (n *NSQD) Main() {
 	if n.opts.StatsdAddress != "" {
 		n.waitGroup.Wrap(func() { n.statsdLoop() })
 	}
+
+	var httpsAddr *net.TCPAddr
+	if n.httpsListener != nil {
+		httpsAddr = n.httpsListener.Addr().(*net.TCPAddr)
+	}
+
+	if n.opts.GossipAddress != "" {
+		serf, err := initSerf(n.opts, n.serfEventChan,
+			n.tcpListener.Addr().(*net.TCPAddr),
+			n.httpListener.Addr().(*net.TCPAddr),
+			httpsAddr,
+			n.broadcastAddr)
+		if err != nil {
+			n.logf("FATAL: failed to initialize Serf - %s", err)
+			os.Exit(1)
+		}
+		n.serf = serf
+	}
+
+	n.waitGroup.Wrap(func() { n.serfEventLoop() })
+	n.waitGroup.Wrap(func() { n.gossipLoop() })
```

Comment:
i'm not intimately familiar enough to know if it's better to put these inside the `GossipAddress` check above. I see `serfEventLoop` does a short circuit check on `self.serf != nil` but don't see anything similar for `gossipLoop`. Feels like they should both have short circuit checks if it doesn't complicate cleanup logic.